### ISSUE TITLE
Nibbler doesnt work properly with leela chess 0 under windows 7 x64

### ISSUE THREAD

**Issue by Vukica14** [rooklift/nibbler#220]
Hello to everyone,
I am happy to have Nibbler and leela chess 0 but v29 of leela doesnt fit with Nibbler  2.4.0 and its working independently when I am running benchmark for example from cmd propmt.

Is there any workaround to help to fix this issue?

Many thanks in advance.
![2023-01-14 16_29_11-Nibbler](https://user-images.githubusercontent.com/93468705/212479746-3f79bab9-6d96-4f1d-b0b0-ad38510e309e.jpg)

**Comment by rooklift** [comment]
Is this different from #203 ?

**Comment by Vukica14** [comment]
It is as I discovered what version of leela is incompatible with nibbler under win7 x64, so I decided to post again.

**Comment by rooklift** [comment]
I still don't know what to suggest.

**Comment by Vukica14** [comment]
This is log file of nibbler:
======================================================================================================================================
Nibbler startup at Sat, 14 Jan 2023 15:48:56 GMT

Launching D:\_LeelaZero\lc0-v0.29.0-windows-gpu-nvidia-cudnn\lc0.exe
(failed) --> uci

**Comment by Vukica14** [comment]
and this is once when I load v26.1

--> uci

Launching D:\_LeelaZero\lc0-v0.26.1-windows-gpu-nvidia-cuda\lc0.exe

(failed) --> quit
--> uci
.        _
. |   _ | |
. |_ |_ |_| v0.26.1 built Jul 15 2020
< id name Lc0 v0.26.1
< id author The LCZero Authors.
< option name WeightsFile type string default <autodiscover>
< option name Backend type combo default cudnn-auto var cudnn-auto var cudnn var cudnn-fp16 var random var check var recordreplay var roundrobin var multiplexing var demux
< option name BackendOptions type string default 
< option name Threads type spin default 2 min 1 max 128
< option name NNCacheSize type spin default 200000 min 0 max 999999999
< option name MinibatchSize type spin default 256 min 1 max 1024
< option name MaxPrefetch type spin default 32 min 0 max 1024
< option name CPuct type string default 2.147000
< option name CPuctAtRoot type string default 2.147000
< option name CPuctBase type string default 18368.000000
< option name CPuctBaseAtRoot type string default 18368.000000
< option name CPuctFactor type string default 2.815000
< option name CPuctFactorAtRoot type string default 2.815000
< option name RootHasOwnCpuctParams type check default true
< option name Temperature type string default 0.000000
< option name TempDecayMoves type spin default 0 min 0 max 100
< option name TempDecayDelayMoves type spin default 0 min 0 max 100
< option name TempCutoffMove type spin default 0 min 0 max 1000
< option name TempEndgame type string default 0.000000
< option name TempValueCutoff type string default 100.000000
< option name TempVisitOffset type string default 0.000000
< option name DirichletNoiseEpsilon type string default 0.000000
< option name DirichletNoiseAlpha type string default 0.300000
< option name VerboseMoveStats type check default false
< option name LogLiveStats type check default false
< option name FpuStrategy type combo default reduction var reduction var absolute
< option name FpuValue type string default 0.443000
< option name FpuStrategyAtRoot type combo default same var reduction var absolute var same
< option name FpuValueAtRoot type string default 1.000000
< option name CacheHistoryLength type spin default 0 min 0 max 7
< option name PolicyTemperature type string default 1.607000
< option name MaxCollisionEvents type spin default 32 min 1 max 65536
< option name MaxCollisionVisits type spin default 9999 min 1 max 1000000
< option name OutOfOrderEval type check default true
< option name MaxOutOfOrderEvalsFactor type string default 1.000000
< option name StickyEndgames type check default true
< option name SyzygyFastPlay type check default true
< option name MultiPV type spin default 1 min 1 max 500
< option name PerPVCounters type check default false
< option name ScoreType type combo default centipawn var centipawn var centipawn_with_drawscore var centipawn_2019 var centipawn_2018 var win_percentage var Q var W-L
< option name HistoryFill type combo default fen_only var no var fen_only var always
< option name MovesLeftMaxEffect type string default 0.100000
< option name MovesLeftThreshold type string default 1.000000
< option name MovesLeftSlope type string default 0.005000
< option name MovesLeftConstantFactor type string default 0.000000
< option name MovesLeftScaledFactor type string default 0.000000
< option name MovesLeftQuadraticFactor type string default 1.000000
< option name DisplayCacheUsage type check default false
< option name MaxConcurrentSearchers type spin default 1 min 0 max 128
< option name DrawScoreSideToMove type spin default 0 min -100 max 100
< option name DrawScoreOpponent type spin default 0 min -100 max 100
< option name DrawScoreWhite type spin default 0 min -100 max 100
< option name DrawScoreBlack type spin default 0 min -100 max 100
< option name NodesPerSecondLimit type string default 0.000000
< option name SolidTreeThreshold type spin default 100 min 1 max 2000000000
< option name SyzygyPath type string default 
< option name Ponder type check default true
< option name UCI_Chess960 type check default false
--> setoption name UCI_Chess960 value true
< option name UCI_ShowWDL type check default false
< option name UCI_ShowMovesLeft type check default false
< option name ConfigFile type string default lc0.config
< option name KLDGainAverageInterval type spin default 100 min 1 max 10000000
< option name MinimumKLDGainPerNode type string default 0.000000
< option name SmartPruningFactor type string default 1.330000
< option name SmartPruningMinimumBatches type spin default 0 min 0 max 10000
< option name NodesAsPlayouts type check default false
< option name RamLimitMb type spin default 0 min 0 max 100000000
< option name MoveOverheadMs type spin default 200 min 0 max 100000000
< option name TimeManager type string default legacy
< option name StrictTiming type check default false
< option name LogFile type string default 
< uciok
--> setoption name LogLiveStats value true
--> setoption name MoveOverheadMs value 0
--> setoption name MultiPV value 500
--> setoption name ScoreType value centipawn
--> setoption name SmartPruningFactor value 0
--> setoption name UCI_ShowWDL value true
--> setoption name VerboseMoveStats value true
--> isready
< readyok
--> ucinewgame
. Found pb network file: D:\_LeelaZero\lc0-v0.26.1-windows-gpu-nvidia-cuda/703350.pb.gz
. Creating backend [cudnn-auto]...
. Switching to [cudnn]...
. CUDA Runtime version: 10.0.0
. Cudnn version: 7.4.2
. Latest version of CUDA supported by the driver: 11.4.0
. GPU: NVIDIA GeForce GTX 850M
. GPU memory: 2 Gb
. GPU clock frequency: 901.5 MHz
. GPU compute capability: 5.0

**Comment by Vukica14** [comment]
Why you close this? maybe someone else has idea.

**Comment by rooklift** [comment]
I closed it because it's identical to #203 which is still open. Still I'll reopen this and close that one, if you prefer.

**Comment by Vukica14** [comment]
Thanks. I prefer to keep this one as it is more detailed.

**Comment by Vukica14** [comment]
its interesting this initial part of log file when I am loading v26.1
Launching D:_LeelaZero\lc0-v0.26.1-windows-gpu-nvidia-cuda\lc0.exe

(failed) --> quit
--> uci
. _
. | _ | |

is that normal behavior?

**Comment by rooklift** [comment]
It seems for 0.29 you are using the `cudnn` version but for 0.26 you were using the `cuda` version - can you try the cuda version of 0.29?

**Comment by rooklift** [comment]
> its interesting this initial part of log file when I am loading v26.1 Launching D:_LeelaZero\lc0-v0.26.1-windows-gpu-nvidia-cuda\lc0.exe
> 
> (failed) --> quit --> uci . _ . | _ | |
> 
> is that normal behavior?

I believe that was it trying to tell the previous engine (which didn't start) to quit, which it can't, because it didn't start.

**Comment by Vukica14** [comment]
since version 26.2 cuda is transfered into cudNN. I try "go infinite" on cuda v29 but it didnt work so before v26.2 it must be CUDA and after that cudNN.
and, thanks for the clarification.

**Comment by rooklift** [comment]
OK, I see. But I don't know what to do.

**Comment by Vukica14** [comment]
I posted also in leela chess but I am still wating for the answer. It seams leela working well but in combo with nibbler making a problems.

**Comment by rooklift** [comment]
btw, in the other issue you mentioned Stockfish not working either. Can you please turn off `Dev --> Show engine state` and then post a screenshot of what it looks like when loading Stockfish?

**Comment by Vukica14** [comment]
sure here is version 15.1
![2023-01-14 17_06_43-Nibbler](https://user-images.githubusercontent.com/93468705/212482056-eb2f87f0-d17e-4b25-b94e-cafca6864690.jpg)

**Comment by rooklift** [comment]
Please turn off `Dev --> Show engine state` first.

**Comment by Vukica14** [comment]
I did, I think
![Uploading 2023-01-14 17_08_07-Nibbler.jpg…]()

**Comment by rooklift** [comment]
OFF

**Comment by rooklift** [comment]
(Or are you saying it doesn't turn off?)

**Comment by Vukica14** [comment]
> (Or are you saying it doesn't turn off?)

sorry I didnt get you. I enabled "enginee state" is there anything else I have to do? But both versions of the stockfish doeasnt work

**Comment by Vukica14** [comment]
and stockfish both versions works well in fritz 16

**Comment by rooklift** [comment]
>I enabled "enginee state"

Please turn it off.

**Comment by Vukica14** [comment]
> > I enabled "enginee state"
> 
> Please turn it off.

here it is
![2023-01-14 17_22_36-Nibbler](https://user-images.githubusercontent.com/93468705/212483006-a0777de2-8dbf-4406-a243-2e284c48f3ea.jpg)

**Comment by Vukica14** [comment]
ooo after GO it works :D

**Comment by rooklift** [comment]
Cool. So does nothing happen when you click "Go" at the top right?

**Comment by rooklift** [comment]
So how else were you trying to start the analysis, and was that other way not working?

**Comment by Vukica14** [comment]
and for leela 29
![2023-01-14 17_23_55-Nibbler](https://user-images.githubusercontent.com/93468705/212483058-162fd269-350a-4d92-b14c-b2d168fb9aad.jpg)

**Comment by Vukica14** [comment]
> So how else were you trying to start the analysis, and was that other way not working?

for versions lower than 26.2 I see that initialization of the enginee (leela) and for higher versions it cant start

**Comment by rooklift** [comment]
Right, yeah, Leela is definitely failing to start. Could it be a memory issue?

I doubt it will help, but try:

`Dev --> Disable hardware acceleration for GUI` (turn it on, ticked, enabled)

**Comment by rooklift** [comment]
And then restart Nibbler.

**Comment by Vukica14** [comment]
> And then restart Nibbler.

I try but it is the same

**Comment by rooklift** [comment]
Can you post your system hardware info?

**Comment by Vukica14** [comment]
> Can you post your system hardware info?

sure just where to find it?

**Comment by rooklift** [comment]
Uh. Well I guess I wanted to know how much RAM your computer has, your graphics card, and how much RAM it has as well.

**Comment by Vukica14** [comment]
here is GPU
 GPU: NVIDIA GeForce GTX 850M
. GPU memory: 2 Gb
. GPU clock frequency: 901.5 MHz
. GPU compute capability: 5.0

and I have 
![2023-01-14 17_29_51-System](https://user-images.githubusercontent.com/93468705/212483380-58186a94-7de2-4f57-9aff-4d2b2f51e3a1.jpg)

**Comment by Vukica14** [comment]
its bit old but I think still powerful 16GB of RAM and quad core

**Comment by rooklift** [comment]
Yeah I think it should be good enough. Uh, can you try the Windows OpenCL version of Leela 0.29, just to see if it works?

**Comment by Vukica14** [comment]
I try and its the same.

**Comment by rooklift** [comment]
When a UCI engine starts up, it sends some info and finally the line `uciok`. The message from Nibbler indicates we never even got this far.

**Comment by rooklift** [comment]
Actually, now that I think about it, the OpenCL client has a substantial "tuning" period on first run, do you see any output at all on the right side of the window?

**Comment by Vukica14** [comment]
> Actually, now that I think about it, the OpenCL client has a substantial "tuning" period on first run, do you see any output at all on the right side of the window?

this is the window 
![Uploading 2023-01-14 17_43_24-Nibbler.jpg…]()
with HW  acc enabled and disabled

**Comment by rooklift** [comment]
OK. Sorry, I wish I knew the cause but I can't figure it out.

**Comment by Vukica14** [comment]
> OK. Sorry, I wish I knew the cause but I can't figure it out.

No prob and many thanks for the wish to help out here.
